lib/wrappers/output.ts

```diff
@@ -0,0 +1,38 @@
+import fs, { PathLike } from "fs";
+import TransformNodeWrapper from "./transform-node";
+import { isAbsolute } from 'path';
+
+const logger = require('heimdalljs-logger')('broccoli:outputWrapper');
+
+const WHITELISTEDOPERATION = new Set([
+  'readFileSync',
+  'existsSync',
+  'lstatSync',
+  'readdirSync',
+  'statSync',
+  'writeFileSync',
+  'appendFileSync',
+  'rmdirSync',
+  'mkdirSync'
+]);
+
+export default function OutputWrapper (node: TransformNodeWrapper) {
+  return new Proxy(fs, {
+    get(target: any, propertyName: string): any {
+      return function() {
+        let [relativePath] = arguments;
+        if (isAbsolute(relativePath)) {
+          throw new Error(`Relative path is expected, path ${relativePath} is an absolute path.`);
+        }
+        let outputPath = node.outputPath + '/' + relativePath;
```

Comment:
What happens when `relativePath` is something like `../../../../etc/hosts`? I think it would "break" the scoping (basically the same issue as absolute paths).